algorithms/map-sparsification/src/keyframe-pruning.cc

```diff
@@ -251,7 +251,9 @@ size_t selectKeyframesBasedOnHeuristics(
   }
 
   // The last vertex is always a keyframe
-  insert_keyframe(current_vertex_id);
+  if (last_keyframe_id != current_vertex_id) {
+    insert_keyframe(current_vertex_id);
```

Comment:
Can you explain why this is something that happens only sometime? Is it because the while loop can terminate either with the last vertex being a keyframe (in which case you do not want to add it again) or the öast vertex is not a keyframe (in which case you want to add it)? In any case, please update the comment in the code to explain this

Reply:
Yes, exactly. You want the first and last vertices to always be keyframes, otherwise you lose information. But before it could happen that you already decided to add the last one in the while loop and then it was being added a second time at the end. This caused a CHECK fail down the code because there's no edge connecting the two vertices obviously.

Reply:
Updated comment to clarify this.